### ISSUE TITLE
Add Image TransformService with Configurable Preprocessing

### ISSUE THREAD

**Issue by kjpou1** [kjpou1/convolution-patterns#2]
**Task Description**:
Implement a reusable `TransformService` for Convolution CV that builds TensorFlow image preprocessing pipelines for training, validation, and inference. Support config-driven behavior via a `transforms_used.yaml` file, and ensure consistency across all pipeline stages.

---

**📌 Goals**:

* Create a `TransformService` class that:

  * Loads transformation settings from YAML
  * Builds a `tf.keras.Sequential` preprocessing pipeline per mode (`train`, `val`, `infer`)
  * Optionally saves the config for reproducibility
* Use `tf.keras.layers` for resizing, normalization, and augmentation
* Integrate with image datasets from `artifacts/data/processed/{split}/...`

---

**📁 Artifacts**:

* `transforms_used.yaml` saved alongside each trained model (under `artifacts/model_name/`)
* Contains image size, rescale factor, and per-mode augmentation flags

---

**🛠️ Implementation Tasks**:

* [ ] Define `TransformService` with:

  * `.from_yaml(path)` to load config
  * `.get_pipeline(mode)` to return a TF preprocessing pipeline
  * `.save_config(path)` to persist config after training
* [ ] Add base YAML config under `configs/transforms_base.yaml`
* [ ] Ensure consistent use of the pipeline across:

  * Training (augmented)
  * Validation (no augmentation)
  * Inference (no augmentation)
* [ ] Add example usage in `train_pipeline.py` and `predict_pipeline.py`

---

**📦 Example Output ????**:

```yaml
# transforms_used.yaml
image_size: [224, 224]
rescale: 1.0 / 255.0

train_augmentation:
  horizontal_flip: true
  rotation: 0.05
  zoom: 0.1
  width_shift: 0.05
  height_shift: 0.05

val_augmentation: false
infer_augmentation: false
```
